lib/_stream_readable.js

```diff
@@ -1070,13 +1073,69 @@ Readable.prototype.wrap = function(stream) {
 };
 
 Readable.prototype[SymbolAsyncIterator] = function() {
-  if (createReadableStreamAsyncIterator === undefined) {
-    createReadableStreamAsyncIterator =
-      require('internal/streams/async_iterator');
+  let stream = this;
+
+  if (typeof stream.read !== 'function') {
+    // v1 stream
+
+    if (!eos) eos = require('internal/streams/end-of-stream');
+
+    const src = stream;
+    stream = new Readable({ objectMode: true }).wrap(src);
+    eos(stream, (err) => destroyImpl.destroyer(src, err));
```

Comment:
I'm confused regarding why we have to add this additional callback - since we call `createAsyncIterator` which calls `destroyImpl` in the `finally` block.

Mind clarifying?

Reply:
Note sure actually, just moved existing code here. I'll investigate.

Reply:
Ah, now I remember. I wrote the original code. We need to destroy the original stream, calling `destroy` on the outer Readable won't destroy the wrapped stream.

Reply:
@benjamingr: Calling destroy on the readable won't destroy the wrapped stream.